components/sup/src/event/nats_message_stream.rs

```diff
@@ -34,74 +83,92 @@ pub struct NatsMessageStream(pub(super) UnboundedSender<NatsMessage>);
-                                connect_method,
-                                server_certificate,
-                                .. } = config;
-
-        let mut client = Client::new(vec![url]);
-
-        // Configure the client connect message
-        client.connect_mut()
-              .await
-              .name(format!("hab_client_{}", supervisor_id))
-              .verbose(true)
-              .token(token.to_string());
-
-        // Configure the tls connector
-        let mut tls_connector = TlsConnector::builder();
-        for certificate in habitat_core::tls::native_tls_wrapper::certificates(None)? {
-            tls_connector.add_root_certificate(certificate);
-        }
-        if let Some(certificate) = server_certificate {
-            tls_connector.add_root_certificate(certificate.into());
-        }
-        let tls_connector = tls_connector.build()?;
-        client.set_tls_config(tls_connector).await;
+
+        println!("NatsMessageStream::new at {}", config.url);
+
+        let token_str = format!("{}", config.token);
+        let name = format!("hab_client_{}", supervisor_id);
+        let cert_path = format!("{}", config.server_certificate.unwrap());
+        let config_url = format!("{}", config.url);
+
+        let nats_options: Options = Options::with_token(&token_str) 
+            .add_root_certificate(&cert_path);
+
+        let mut nats_client: NatsClient = NatsClient::new(&nats_options, &config_url);
+
+        let (tx, rx) = futures_mpsc::unbounded::<NatsMessage>();
 
         // Connect to the server. If a timeout was set, we want to ensure we establish a connection
         // before exiting the function. If we do not connect within the timeout we return an error.
         // If we do not have a timeout, we dont care if we can immediately connect. Instead we spawn
         // a future that will resolve when a connection is possible. Once we establish a
         // connection, the client will handle reconnecting if necessary.
-        if let Some(timeout) = connect_method.into() {
-            time::timeout(timeout, client.connect()).await
-                                                    .map_err(|_| Error::ConnectNatsServer)?;
-        } else {
-            let client = Client::clone(&client);
-            tokio::spawn(async move { client.connect().await });
+
+        if let Some(timeout) = config.connect_method.into() {
+            println!("About to connect with timeout");
+            let res = time::timeout(timeout, nats_client.connect())
+                .await
+                .map_err(|_| Error::ConnectNatsServer)?;
+            println!("result from timeout = {:?}", res);
+        }
+        else {
+            println!("About to connect - no timeout specified");
+            let nats_opts: Options = Options::with_token(&token_str) 
+                .add_root_certificate(&cert_path);
+            tokio::spawn(async move { nats_client.connect().await });
```

Comment:
Not sure why this comment disappeared before!

Basically I think you can do with the `NatsClient` and replace the above if/else with:

```
        let connection = match config.connect_method.into() {
            Some(timeout) => {
                println!("About to connect with timeout");
                time::timeout(timeout, nats_options.connect(&config_url))
                    .await
                    .map_err(|_| Error::ConnectNatsServer)?.unwrap()
            }
            None => {
                println!("About to connect - no timeout specified");
                let nats_opts: Options = Options::with_token(&token_str) 
                    .add_root_certificate(&cert_path);
                nats_options.connect(&config_url).await.unwrap()
            }
        };
```